tla/Network.tla

```diff
@@ -0,0 +1,86 @@
+-------------------------------- MODULE Network -------------------------------
+EXTENDS ccfraft, Sequences
+
+----------------------------------------------------------------------------------
+\* Reordering and duplication of messages:
+
+ReorderDupInitMessageVar ==
+    messages = <<>>
+    
+ReorderDupWithMessage(m, msgs) == 
+    IF m \notin (DOMAIN msgs) THEN
+        msgs @@ (m :> 1)
+    ELSE
+        [ msgs EXCEPT ![m] = @ + 1 ]
+
+ReorderDupWithoutMessage(m, msgs) == 
+    IF msgs[m] = 1 THEN
+        [ msg \in ((DOMAIN msgs) \ {m}) |-> msgs[msg] ]
+    ELSE
+        [ msgs EXCEPT ![m] = @ - 1 ]
+
+ReorderDupMessages ==
+    DOMAIN messages
+
+ReorderDupMessagesOf(s) ==
+    { m \in Messages : m.dest = s }
+
+----------------------------------------------------------------------------------
+\* Reordering and deduplication of messages (iff the spec removes message m from
```

Comment:
Would it make any sense to use a set per destination instead of one overall set? I am just wondering if it would make traces easier to read as it can be tricky when messages is large

Reply:
I'd suggest to use a [TLC alias](https://github.com/microsoft/CCF/blob/7ae7f7acc16ba870a1f8a52070bef7f5a3c0b021/tla/Traceccfraft.tla#L415-L456) to make traces more readable:

```tla
MyAlias ==
[
  ..., 
  otherVariable |-> otherVariable,
  messagesPerServer |-> [ s \in Servers |-> { m \in Messages: m.dest = s } ]
]
```

```
SPECIFICATION Spec
...
ALIAS MyAlias
```

Reply:
https://github.com/microsoft/CCF/pull/5651/commits/785a18dcf4070797b2a301d1597727390545bacf#diff-8eff7eda246f4f97e2efbcd8bc2a1ad805688764ab11afdbf9fa31b4f80d492cR1402